### PR TITLE
Use hosts filters instead of hosts

### PR DESCRIPTION
**Pull request by wdkcc**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #
 
**Special notes for your reviewer**:

### CODE REVIEW

File: center/router/router_task.go [hunk 1]
```diff
@@ -167,11 +165,23 @@ func (rt *Router) taskAdd(c *gin.Context) {
 	var f taskForm
 	ginx.BindJSON(c, &f)
 
+	query := models.GetHostsQuery(f.HostsQuery)
+	// set limit 0,0 to get all hosts
+	targets, err := models.TargetGetsByFilter(rt.Ctx, query, 0, 0, false)
+	ginx.Dangerous(err)
+	if len(targets) == 0 {
+		ginx.Dangerous(errors.New(fmt.Sprintf("no targets found by %v", f.HostsQuery)))
```
Comment (710leo) [inline center/router/router_task.go:173]: ginx.Bomb

File: models/task_tpl.go [hunk 2]
```diff
@@ -171,38 +172,33 @@ func (t *TaskTpl) Save(ctx *ctx.Context, hosts []string) error {
...
+	err := DB(ctx).Table("task_tpl_host").Where("id=?", t.Id).Order("ii").Pluck("host", &hosts).Error
+	if err != nil {
+		return nil, err
+	}
 
-			if err != nil {
-				return err
-			}
-		}
+	targets, err := TargetGetsByHosts(ctx, hosts)
```
Comment (710leo) [inline models/task_tpl.go:185]: 直接用 hosts 就可以了

File: models/migrate/migrate.go [hunk 3]
```diff
@@ -150,6 +151,11 @@ type ChartShare struct {
 type TaskRecord struct {
 	EventId int64 `gorm:"column:event_id;bigint(20);not null;default:0;comment:event id;index:idx_event_id"`
 }
+
+type TaskTpl struct {
+	HostsQuery []models.HostQuery `json:"hosts_query" gorm:"serializer:json"`
```
Comment (710leo) [inline models/migrate/migrate.go:156]: 补充下 varchar

File: center/router/router_task.go (hunk 1 above)
Comment (wdkcc) [reply center/router/router_task.go:173]: 已修改

File: models/migrate/migrate.go (hunk 3 above)
Comment (wdkcc) [reply models/migrate/migrate.go:156]: 已补充

File: models/task_tpl.go (hunk 2 above)
Comment (wdkcc) [reply models/task_tpl.go:185]: 已修改

**Pull request merged**